autoscheduler/frontend/src/tests/AvailabilityUI.test.tsx

```diff
@@ -0,0 +1,360 @@
+import * as React from 'react';
+import {
+  render, fireEvent,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import autoSchedulerReducer from '../redux/reducers';
+import Schedule from '../components/Schedule/Schedule';
+import * as styles from '../components/Schedule/Schedule.css';
+
+describe('Availability UI', () => {
+  afterEach(() => {
+    document.body.innerHTML = '';
+  });
+
+  describe('adds availability cards', () => {
+    test('with a single click', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer);
+      const {
+        getByText, getByLabelText, queryByText, queryByLabelText,
+      } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+      const mouseEventProps = {
+        button: 0,
+        clientY: 500,
+      };
+      // should create a 30-minute block starting at mid-day, or 14:30
+      const expectedStart = '14:30';
+      const expectedEnd = '15:00';
+
+      // mocks height of the calendar day as 1000
+      const dayZero = document.getElementsByClassName(styles.calendarDay)[0];
+      jest.spyOn(dayZero, 'clientHeight', 'get')
+        .mockImplementation(() => 1000);
+
+      // act
+      const monday = getByLabelText('Monday');
+      fireEvent.mouseDown(monday, mouseEventProps);
+      fireEvent.mouseUp(monday, mouseEventProps);
+
+      // assert
+      expect(getByText('BUSY')).toBeInTheDocument();
+      // ensures that all times are actually calculated based on mocked measurements
+      expect(queryByText(/NaN/)).not.toBeInTheDocument();
+      expect(queryByLabelText('Adjust Start Time'))
+        .toHaveAttribute('aria-valuetext', expectedStart);
+      expect(queryByLabelText('Adjust End Time'))
+        .toHaveAttribute('aria-valuetext', expectedEnd);
+    });
+
+    test('with dragging', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const {
+        getByText, getByLabelText, queryByText, queryByLabelText,
+      } = render(
+        <Provider store={store}>
+          <Schedule />
+        </Provider>,
+      );
+      const startEventProps = {
+        button: 0,
+        clientY: 500,
```

Comment:
Can you abstract this into a function to convert a time into clientY (similar to `eventToTime` in schedule but the other way around)? It would make it more straightforward to tell what times are being dragged rather than just looking at the expected start/end times

Reply:
Good idea, I'll try to get it done after classes tonight